components/ArticleCard.js

```diff
@@ -1,6 +1,15 @@
 import styled from "styled-components";
 import Image from "next/image";
 
-export default function ArticleCard({ imageUrl }) {
-  return <Image src={imageUrl} width={600} height={400}></Image>;
+export default function ArticleCard({ text, showContent }) {
+  return <StyledCard onClick={showContent}>{text}</StyledCard>;
```

Comment:
Instead of working with `onClick` I recommend you to work with [`next/link`](https://nextjs.org/docs/api-reference/next/link) to implement your dynamic routing as this is the more semantic use case. [See the next.js documentation](https://nextjs.org/docs/routing/dynamic-routes) on how to implement dynamic routing. If you want to link to another page you won't use a `section` element, but a `link` element.
I also found [this repo in your course channel about Next.js Routing by Thomas](https://github.com/ThomasF85/cgn-web-22-4-fish-shop). Maybe this example is helpful for you.